pymesos/process.py

```diff
@@ -202,6 +202,11 @@ def on_event(self, event):
     def on_close(self):
         raise NotImplementedError
 
+    def process_event(self, event):
+        with self._lock:
+            if not self._stop:
+                self.on_event(event)
```

Comment:
Should we block other threads when `on_event` is running? the `on_event` is not locked before.